### PR TITLE
Do a better job of picking ports, use sauce-legal ports

### PR DESCRIPTION
**Pull request by rictic**
- [x] CHANGELOG.md has been updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## Unreleased
+
+### Fixed
+* When no port is given, do a better job of finding an available port.
+ * Uses the list of [sauce-legal](https://wiki.saucelabs.com/display/DOCS/Sauce+Connect+Proxy+FAQS#SauceConnectProxyFAQS-CanIAccessApplicationsonlocalhost?) ports.
```
Comment (justinfagnani) [inline CHANGELOG.md]: fix leading spacing

File: src/start_server.ts [hunk 2]
```diff
@@ -86,7 +85,7 @@ async function applyDefaultOptions(options: ServerOptions):
     Promise<ServerOptions> {
       const withDefaults = Object.assign({}, options);
       Object.assign(withDefaults, {
-        port: await nextOpenPort(options.port),
+        port: options.port || 0,
```
Comment (justinfagnani) [inline src/start_server.ts]: This function could be sync now. Probably being async to choose a port wasn't great to begin with.

Comment (rictic) [inline src/start_server.ts]: Done

File: CHANGELOG.md (hunk 1 above)
Comment (rictic) [inline CHANGELOG.md]: Leading space was intended, it's a sub-point of the above

**Comment by rictic** [comment]
Fixes #134

**Pull request merged**